pkg/webhook/workspace/handler/immutable.go

```diff
@@ -81,8 +93,26 @@ func (h *WebhookHandler) handleImmutableRoute(oldObj, newObj runtime.Object, use
 	if username == h.ControllerSAName {
 		return true, ""
 	}
-	if reflect.DeepEqual(oldObj, newObj) {
+	if changePermitted(oldObj, newObj) {
 		return true, ""
 	}
 	return false, "object is owned by workspace and cannot be updated."
 }
+
+func changePermitted(oldObj, newObj runtime.Object) bool {
+	oldCopy := oldObj.DeepCopyObject()
+	newCopy := newObj.DeepCopyObject()
+	oldMeta, ok := oldCopy.(metav1.Object)
+	if !ok {
+		log.Error(fmt.Errorf("Object is not a valid k8s object: does not have metadata"), "Failed to compare objects")
```

Comment:
Does it make sense to add at least object kind to logs?

Reply:
Done